lib/stages/ReadGain.cpp

```diff
@@ -115,28 +119,20 @@ bool ReadGain::update_gains_frb_callback(nlohmann::json& json) {
     return true;
 }
 
-bool ReadGain::update_gains_tracking_callback(nlohmann::json& json) {
-    if (update_gains_tracking) {
-        WARN("[TRACKING] cannot handle two back-to-back gain updates, rejecting the latter");
-        return true;
-    }
-    try {
-        _gain_dir_tracking = json.at("tracking_gain_dir").get<std::vector<std::string>>();
-        std::string output_msg = "[TRACKING] Updating gains from ";
-        for (int i = 0; i < _num_beams; i++) {
-            output_msg += _gain_dir_tracking[i];
-            output_msg += " ";
-        }
-        INFO("{:s}", output_msg);
-    } catch (std::exception const& e) {
-        WARN("[Tracking Beamformer] Fail to read gain_dir {:s}", e.what());
-        return false;
-    }
+bool ReadGain::update_gains_tracking_callback(nlohmann::json& json, const uint8_t beam_id) {
     {
         std::lock_guard<std::mutex> lock(mux);
+        try {
+            _gain_dir_tracking.push(
+                std::make_pair(beam_id, json.at("gain_dir").get<std::string>()));
+            INFO("[Tracking Beamformer] Updating gains from {:s}",
+                 _gain_dir_tracking.back().second);
+        } catch (std::exception const& e) {
+            WARN("[Tracking Beamformer] Fail to read gain_dir {:s}", e.what());
+            return false;
+        }
         update_gains_tracking = true;
     }
-    cond_var.notify_all();
```

Comment:
You still need this notify, otherwise the loop in the main thread won't wake up to call `read_gain_tracking()`.